crates/iota-rest-api/src/response.rs

```diff
@@ -130,18 +130,26 @@ pub async fn append_info_headers(
     State(state): State<RestService>,
     response: Response,
 ) -> impl IntoResponse {
-    let latest_checkpoint = state.reader.inner().get_latest_checkpoint().unwrap();
+    let (epoch, sequence_number, timestamp_ms) = match state.reader.inner().get_latest_checkpoint()
+    {
+        Ok(checkpoint) => (
+            checkpoint.epoch(),
+            *checkpoint.sequence_number(),
+            checkpoint.timestamp_ms,
+        ),
+        Err(_) => (0, 0, 0),
```

Comment:
Instead of ignoring the error, it might be better to return it from this function. Overriding the error with (0,0,0) doesn't seem good to me, also because it might be an important error from the state reader.